android/src/main/java/com/android/simplSdk/SimplSdkModule.java

```diff
@@ -26,61 +26,96 @@ public SimplSdkModule(ReactApplicationContext reactContext) {
         super(reactContext);
     }
 
-    @Override
-    public String getName() {
-        return "SimplSdk";
+    @ReactMethod
+    public void isUserApproved(final ReadableMap map, final Callback successCallback, final Callback errorCallback){
+        final ReadableMap paramsMap = map.getMap("params");
+        try {
+            Simpl.init(getReactApplicationContext(), map.getString("merchantId"));
+            if (map.getBoolean("isSandbox"))
+                Simpl.getInstance().runInSandboxMode();
+
+            SimplUser user = new SimplUser(map.getString("email"), map.getString("phone_number"));
+            Simpl.getInstance().isUserApproved(user)
+                .addParam("transaction_amount_in_paisa", paramsMap.getString("amount_in_paisa"))
+                .execute(new SimplUserApprovalListenerV2() {
+                        @Override
+                        public void onSuccess(final boolean isUserApproved, String buttonText, boolean showSimplIntroduction){
+                            successCallback.invoke(isUserApproved);
+                        }
+
+                        @Override
+                        public void onError(Throwable throwable){
+                            errorCallback.invoke(throwable.getLocalizedMessage());
+                        }
+                    });
+        } catch(Exception ex){
+            errorCallback.invoke(ex.getMessage());
+        }
+
     }
 
     @ReactMethod
     public void isApproved(final String merchantId, final String mobileNumber, final String emailId, final boolean isSandbox,
                            final Callback successCallback, final Callback errorCallback) {
-        Simpl.init(getReactApplicationContext(), merchantId);
-        Log.d(TAG, "isApproved(): merchantId: " + merchantId + " mobileNumber: " + mobileNumber + " emailId: " + emailId);
-        if (isSandbox)
-            Simpl.getInstance().runInSandboxMode();
-        Simpl.getInstance().isUserApproved(new SimplUser(emailId, mobileNumber))
+        try {
+            Simpl.init(getReactApplicationContext(), merchantId);
+            Log.d(TAG, "isApproved(): merchantId: " + merchantId + " mobileNumber: " + mobileNumber + " emailId: " + emailId);
+            if (isSandbox)
+                Simpl.getInstance().runInSandboxMode();
+            Simpl.getInstance().isUserApproved(new SimplUser(emailId, mobileNumber))
                 .execute(new SimplUserApprovalListenerV2() {
+                        @Override
+                        public void onSuccess(final boolean b, String s, boolean b1) {
+                            successCallback.invoke(b);
+                        }
+
+                        @Override
+                        public void onError(Throwable throwable) {
+                            errorCallback.invoke(throwable.getLocalizedMessage());
+                        }
+                    });
+        }catch(Exception ex){
+            errorCallback.invoke(ex.getMessage());
```

Comment:
In some places we use `getMessage` and in some places we use `getLocalizedMessage`. Lets make it consistent and use the preferred one?